pyfr/plugins/tavg.py

```diff
@@ -147,6 +170,64 @@ def _eval_fun_exprs(self, intg, accex):
         # Stack up the expressions for each element type and return
         return [np.dstack(exs).swapaxes(1, 2) for exs in exprs]
 
+    def _eval_fun_var(self, dev, accex):
+        dfexpr = []
+        eps = self.eps
+        an = self.anames
+        fexp = self.fexprs
+        devs = [d.swapaxes(0,1) for d in dev]
+        
+        # Iterate over each element type our averaging region
+        for avals in accex:
+            df = []
+            av = avals.swapaxes(0,1)
+            
+            # Iterate over averaged variables
+            for idx,avar in enumerate(an):
+                h = np.zeros_like(av, dtype=np.float64)
+
+                # Calculate step size  
+                h[idx] = eps * av[idx]
+                h[idx][h[idx] == 0] = eps
+
+                # Prepare the substitution dictionary
+                subsh = dict(zip(an, av+h))
+                subs = dict(zip(an, av))
+            
+                # Calculate derivatives for functional averages
+                df.append([(npeval(v, subsh) - npeval(v, subs)) / h[idx]
+                             for v in fexp])
+            
+            dfexpr.append(np.stack(df).swapaxes(0,1))
+        
+        # Multiply with variance and take the RMS value
+        return [np.sqrt(np.einsum('ij..., j... -> i...', 
+            df**2, sd**2)).swapaxes(0,1) for df, sd in zip(dfexpr, devs)]
+    
+    def _acc_avg_var(self, intg, currex, dowrite, doaccum):    
+        prevex = self.prevex
+        vaccex = self.vaccex
+        accex =  self.accex
+                
+        # Weights for online variance and average
+        Wmp1mpn = (intg.tcurr - self.prevt)
+        W1mpn = (intg.tcurr - self.tstart_acc)
+        W1m = self.W1m
+
+        # Iterate over element type
+        for v, a, p, c in zip(vaccex, accex, prevex, currex):
+
+            # Accumulate online variance
+            v += Wmp1mpn*(c ** 2 + p** 2) - 0.5*Wmp1mpn*(p+ c)**2 + \
+                self.sw*((W1m / (2 * Wmp1mpn * W1mpn)) * \
```

Comment:
Use an if statement here; also should be no need for storing sw as we can figure it out.